src/operator/tensor/matrix_op-inl.h

```diff
@@ -32,18 +32,18 @@ struct ReshapeParam : public dmlc::Parameter<ReshapeParam> {
     DMLC_DECLARE_FIELD(shape)
     .set_default(nnvm::Tuple<int>())
     .describe("The target shape");
-    DMLC_DECLARE_FIELD(keep_highest).set_default(false)
-    .describe("(Deprecated! Use ``shape`` instead.) Whether keep the highest dim unchanged."
-              "If set to true, then the first dim in target_shape is ignored,"
-              "and always fixed as input");
     DMLC_DECLARE_FIELD(reverse)
     .set_default(false)
-    .describe("If true then translating the input shape from right to left");
+    .describe("If true then the special values are inferred from right to left");
     DMLC_DECLARE_FIELD(target_shape)
     .set_default(TShape(tmp, tmp + 2))
     .describe("(Deprecated! Use ``shape`` instead.) "
               "Target new shape. One and only one dim can be 0, "
               "in which case it will be inferred from the rest of dims");
+    DMLC_DECLARE_FIELD(keep_highest).set_default(false)
+    .describe("(Deprecated! Use ``shape`` instead.) Whether keep the highest dim unchanged."
+              "If set to true, then the first dim in target_shape is ignored,"
+              "and always fixed as input");
   }
```

Comment:
Please don't change the order of of fields declared by `DMLC_DECLARE_FIELD `. `DMLC_DECLARE_FIELD(keep_highest).set_default(false)` should be followed by `DMLC_DECLARE_FIELD(shape)`.
The order determines the auto-generated cpp interface for the operator. This change breaks the cpp-package build.

@nswamy I'm going to make a PR to revert this change, and add a CI job for cpp-package.
Also ping @zackchase @Roshrini @jiajiechen  for awareness.